### PR TITLE
ci, Add vendored-go to Makefile

### PR DESCRIPTION
**Pull request by RamLavi**
**What this PR does / why we need it**:
Currently, if go is not installed on the host
then e2e tests will fail.
We want to be independent of this.
Added install of vendored go, according to
the version in go.mod

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by maiqueb** [comment]
/lgtm
/hold

I'll have to defer to @phoracek since he's a lot more versed in the go tooling ecosystem than I am.

**Review by phoracek — APPROVED** [review]
/approve
/hold cancel

Looks fine and matches what we have in other projects.

**Pull request merged**
